test/snippet/io/alignment_file/alignment_file_output.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+
+#include <seqan3/alphabet/nucleotide/all.hpp>
+#include <seqan3/io/alignment_file/all.hpp>
+#include <seqan3/io/sequence_file/all.hpp>
+#include <seqan3/range/view/persist.hpp>
+
+using namespace seqan3;
+
+int main()
+{
+
+auto tmp_dir = std::filesystem::temp_directory_path();
+
+// The output snippets will create a /tmp/my.sam
+{
+//! [filename_construction]
+alignment_file_output fout{tmp_dir/"my.sam"}; // SAM format detected, std::ofstream opened for file
+//! [filename_construction]
+}
+
+{
+//! [format_construction]
+// no need to specify the template arguments <...> for format specialization:
+alignment_file_output fout{tmp_dir/"my.sam", fields<field::MAPQ>{}};
+//! [format_construction]
+}
+
+{
+//! [write_range]
+alignment_file_output fout{tmp_dir/"my.sam"};
+
+std::vector<std::tuple<dna5_vector, std::string>> range
+{
+    { "ACGT"_dna5, "First" },
+    { "NATA"_dna5, "2nd" },
+    { "GATA"_dna5, "Third" }
+}; // a range of "records"
+
+fout = range; // will iterate over the records and write them
+//! [write_range]
+}
+
+{
+//! [set_header]
+alignment_file_output fout{tmp_dir/"my.sam"};
+
+// add information to the header of the file.
+fout.header().comments.push_back("This is a comment");
+//! [set_header]
+}
+
+// \todo TODO: uncomment once seqan3::alignment_file_input is implemented.
+#if 0
+{
+//! [custom_fields]
+alignment_file_input  fin{"input.sam", fields<field::REF_OFFSET, field::FLAG>{}};
+alignment_file_output fout{"output.sam"}; // doesn't have to match the configuration
+
+for (auto & r : fin)
+{
+    fout.push_back(r); // copy all the records.
+}
+//! [custom_fields]
+}
+#endif
+
+// Create a /tmp/input.fasta for the snippet below
+{
+    sequence_file_output fout{tmp_dir/"input.fasta"};
+    fout.emplace_back("ACGT"_dna4, "TEST1");
+    fout.emplace_back("AGGCTGA"_dna4, "Test2");
+    fout.emplace_back("ACTGA"_dna4, "Test2");
+}
+// Create a /tmp/input.sam for the snippet below
+{
+    alignment_file_output fout{tmp_dir/"input.sam"};
+
+    std::vector<std::tuple<dna5_vector, std::string>> range
+    {
+        { "ACGT"_dna5, "First" },
+        { "NATA"_dna5, "2nd" },
+        { "GATA"_dna5, "Third" }
+    }; // a range of "records"
+
+    fout = range; // will iterate over the records and write them
+}
+
+{
+//! [input_range]
+// file format conversion in one line:
+alignment_file_output{"/tmp/output.sam"} = sequence_file_input{"/tmp/input.fasta"};
```

Comment:
Shouldn't this example be 
```cpp
alignment_file_output{"/tmp/output.sam"} = alignment_file_input{"/tmp/input.bam"};
```

i.e. conversion between two formats of the same file type. I mean in the above example, it actually works to transforma sequence_file to an alignment_file, but it doesn't really make a lot of sense.

Also you still use `/tmp` here and not a real temporary directory.

Reply:
well maybe FASTA -> SAm does not make sense but SAM -> FASTA actually does. There is even a samtools command especially for this. But of course coversion within the same type makes more sense in the first place... we just don't have another format :trollface:

Reply:
> SAM -> FASTA actually does. There is even a samtools command especially for this. 

Yes, but that's why we have the sequence_file_format_sam in addition to te alignment_file_format_sam, or not?